Alloy/Encoders/MaxEncoder.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  MaxEncoder.swift
+//  Adjuster
+//
+//  Created by Eugene Bokhan on 13/02/2019.
+//
+
+import Metal
+import simd
+
+final public class MaxEncoder {
+
+    enum Errors: Error {
+        case functionCreationFailed
+        case encoderCreationFailed
+        case invalidBufferStorageMode
+    }
+
+    let device: MTLDevice
+    let library: MTLLibrary
+    let pipelineState: MTLComputePipelineState
+
+    public init(library: MTLLibrary) throws {
+        self.device = library.device
+        self.library = library
+
+        guard
+            let function = library.makeFunction(name: MaxEncoder.functionName)
+        else { throw Errors.functionCreationFailed }
+        let pipelineState = try device.makeComputePipelineState(function: function)
+        self.pipelineState = pipelineState
+    }
+
+    public func encode(inputTexture: MTLTexture,
+                       blockSize: BlockSize,
+                       resultBuffer: MTLBuffer,
+                       in commandBuffer: MTLCommandBuffer) throws {
+        guard
+            let encoder = commandBuffer.makeComputeCommandEncoder()
```

Comment:
this is not an alloy syntax

Reply:
@s1ddok what do you exactly mean?

Reply:
@eugenebokhan I adked you to use existing extensions in Alloy, command buffer has .compute {